src/jit/morph.cpp

```diff
@@ -10998,6 +10998,7 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                 //
                 // a % b = a - (a / b) * b;
                 //
+                assert(!optValnumCSE_phase);
```

Comment:
It would be useful to add a comment to that effect.

---
In reply to: [94484298](https://github.com/dotnet/coreclr/pull/8789#discussion_r94484298) [](ancestors = 94484298)